pkg/blockstorage/awsebs/awsebs.go

```diff
@@ -407,13 +408,13 @@ func (s *ebsStorage) VolumeCreateFromSnapshot(ctx context.Context, snapshot bloc
 	if snapshot.Volume.VolumeType == "" || snapshot.Volume.Az == "" || snapshot.Volume.Tags == nil {
 		return nil, errors.Errorf("Required volume fields not available, volumeType: %s, Az: %s, VolumeTags: %v", snapshot.Volume.VolumeType, snapshot.Volume.Az, snapshot.Volume.Tags)
 	}
-
-	zones, err := zone.FromSourceRegionZone(ctx, s, snapshot.Region, snapshot.Volume.Az)
+	cli, _ := kube.NewClient()
+	zones, err := zone.FromSourceRegionZone(ctx, s, cli, snapshot.Region, snapshot.Volume.Az)
 	if err != nil {
 		return nil, err
 	}
 	if len(zones) != 1 {
-		return nil, errors.Errorf("Length of zone slice should be 1, got %d", len(zones))
+		log.Debug().Print("Length of zone slice is greater than 1", field.M{"numberOfZones": len(zones)})
```

Comment:
What motivates this change in behavior?

Reply:
The scenario where, if the original zone wasn't found in the cluster we would return a list of all available zones in the cluster.

Reply:
Is the order of that list deterministic? Does that mean we would end up using the same node since below we use zones[0]?